src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -872,12 +872,24 @@ private static bool CheckNotNamespaceOrType(BoundExpression expr, BindingDiagnos
             }
         }
 
-        private bool CheckLocalValueKind(SyntaxNode node, BoundLocal local, BindValueKind valueKind, bool checkingReceiver, BindingDiagnosticBag diagnostics)
+        private void CheckAddressOfInAsyncOrIteratorMethod(SyntaxNode node, BindValueKind valueKind, BindingDiagnosticBag diagnostics)
         {
-            if (valueKind == BindValueKind.AddressOf && this.IsInAsyncMethod())
+            if (valueKind == BindValueKind.AddressOf)
             {
-                Error(diagnostics, ErrorCode.WRN_AddressOfInAsync, node);
+                if (this.IsInAsyncMethod())
+                {
+                    Error(diagnostics, ErrorCode.WRN_AddressOfInAsync, node);
+                }
+                else if (this.IsDirectlyInIterator && Compilation.IsFeatureEnabled(MessageID.IDS_FeatureRefUnsafeInIteratorAsync))
+                {
+                    Error(diagnostics, ErrorCode.ERR_AddressOfInIterator, node);
```

Comment:
>Error(diagnostics, ErrorCode.ERR_AddressOfInIterator, node); [](http://example.com/codeflow?start=20&length=60)

Is this code going to handle a case of taking address of a field of a local?

Reply:
Yes, see `Iterator_UnsafeBlock_Field_03`.